File: src/main/java/com/app/service/UnitService.java
package com.app.service;

import com.app.domain.Unit;
import com.app.domain.enumeration.UnitType;
import java.util.List;

public interface UnitService {
    /**
     * Create/Update an unit
     * @param unit
     * @return unit entity
     */
    Unit save(Unit unit);
    /**
     * Delete an unit by id
     * @param id : id of unit want to delete
     */
    void delete(Long id);
    /**
     * Find all unit by type
     * @param type: type of unit (BOMON,PHONGTHINGHIEM)
     * @return a list of unit
     */
    List<Unit> findByType(UnitType type);
    /**
     * Find unit by name
     * @param key: part of unit's name
     * @return list units
     */
    List<Unit> findByName(String key);
}

File: src/main/webapp/app/entities/unit/unit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpResponse } from '@angular/common/http';
import { Observable } from 'rxjs';

import { SERVER_API_URL } from 'app/app.constants';
import { createRequestOption } from 'app/shared';
import { IUnit, UnitType } from 'app/shared/model/unit.model';

type EntityResponseType = HttpResponse<IUnit>;
type EntityArrayResponseType = HttpResponse<IUnit[]>;

@Injectable({ providedIn: 'root' })
export class UnitService {
  public resourceUrl = SERVER_API_URL + 'api/units';
  public findByTypeUrl = SERVER_API_URL + 'api/units-by-type';
  public findByNameUrl = SERVER_API_URL + 'api/units-by-name';

  constructor(protected http: HttpClient) {}

  create(unit: IUnit): Observable<EntityResponseType> {
    return this.http.post<IUnit>(this.resourceUrl, unit, { observe: 'response' });
  }

  update(unit: IUnit): Observable<EntityResponseType> {
    return this.http.put<IUnit>(this.resourceUrl, unit, { observe: 'response' });
  }

  find(id: number): Observable<EntityResponseType> {
    return this.http.get<IUnit>(`${this.resourceUrl}/${id}`, { observe: 'response' });
  }

  query(req?: any): Observable<EntityArrayResponseType> {
    const options = createRequestOption(req);
    return this.http.get<IUnit[]>(this.resourceUrl, { params: options, observe: 'response' });
  }

  delete(id: number): Observable<HttpResponse<any>> {
    return this.http.delete<any>(`${this.resourceUrl}/${id}`, { observe: 'response' });
  }

  findByType(type: UnitType) {
    return this.http.get<IUnit[]>(`${this.findByTypeUrl}/${type}`, { observe: 'response' });
  }
  findByName(key: String) {
    return this.http.get<IUnit[]>(`${this.findByNameUrl}/${key}`, { observe: 'response' });
  }
}
